src/mesh/index_derivs.cxx

```diff
@@ -872,6 +866,7 @@ const Field2D Mesh::applyYdiff(const Field2D &var, Mesh::deriv_func func, CELL_L
 
   Field2D result(this);
   result.allocate(); // Make sure data allocated
+  result.setLocation(diffloc);
```

Comment:
This is different to the X functions which now use the passed `loc` rather than `diffloc` -- is there a reason for the difference?

Reply:
It was different because staggering was not implemented for applyYdiff(Field2D). I've implemented it now in 2a35d4d.